components/ciphertool/src/main/java/org/wso2/ciphertool/CipherTool.java

```diff
@@ -390,8 +390,10 @@ private static void updateDeploymentConfigurationWithEncryptedKeys(Map<String, S
                                                                                   Constants.KEY_VALUE_SEPERATOR);
                             if (stringTokenizer.hasMoreTokens()) {
                                 String key = stringTokenizer.nextToken();
-                                String value = encryptedKeyMap.get(key.trim());
-                                line = key.concat(" = \"").concat(value).concat("\"");
+                                if (encryptedKeyMap.containsKey(key.trim())) {
+                                    String value = encryptedKeyMap.get(key.trim());
+                                    line = key.concat(" = \"").concat(value).concat("\"");
+                                }

```

Comment:
Shall we print a meaningful error in else part?

Reply:
-1. Here we're checking whether the key exist in the Map. IMO it not necessary to print an message.